api_app/analyzers_manager/observable_analyzers/onionscan.py

```diff
@@ -0,0 +1,49 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+from api_app.analyzers_manager.classes import DockerBasedAnalyzer, ObservableAnalyzer
+from tests.mock_utils import patch
+
+
+class Onionscan(ObservableAnalyzer, DockerBasedAnalyzer):
+    name: str = "Onionscan"
+    url: str = "http://tor_analyzers:4004/onionscan"
+    # http request polling max number of tries
+    max_tries: int = 1000
+    # interval between http request polling (in seconds)
+    poll_distance: int = 30
```

Comment:
In your experience with using onionscan, did u notice it taking a lot of time to return result ? I ask because you've made max_tries 1000 and poll_distance 30 so that's 30,000 seconds or 8 hours :smiling_face_with_tear:

Reply:
I'd ask you to run 2-3 scans and notice the average time it takes onionscan to return result.

Reply:
Ideally, the values should be max_tries = 30, poll_distance = 10. So that is 300 seconds or 5 minutes. Which means the `soft_time_limit` should be made ~330.

Reply:
> In your experience with using onionscan, did u notice it taking a lot of time to return result ? I ask because you've made max_tries 1000 and poll_distance 30 so that's 30,000 seconds or 8 hours smiling_face_with_tear

Yikes, I added that for testing, seems I forgot to remove it.

> I'd ask you to run 2-3 scans and notice the average time it takes onionscan to return result.

> Ideally, the values should be max_tries = 30, poll_distance = 10. So that is 300 seconds or 5 minutes. Which means the `soft_time_limit` should be made ~330.

I have noticed it takes more than 10 minutes, I'll take an average and try if it works out.

Reply:
I just checked It took more than an hour, and celery just timed it out.

I'm leaving 3 consecutive ones running on a ssh server for a day and I'll return the results tomorrow.

Reply:
So, it looks like onionscan has a problem with v3 addresses, I checked out its issues and someone had a patch for it (onionscan#162) I added the patch and refactored the source, and it looks like its working. I'm now running tests on the ssh shell and will take an average for the timeouts :)

Reply:
> I checked out its issues and someone had a patch for it (onionscan#162) I added the patch and refactored the source, and it looks like its working.

Really appreciate you going the extra mile with the effort.

So I guess we can make the soft_time_limit and the max_tries x poll_distance to be about 15 minutes.

Reply:
Sure Sir 👍